ecommerce_extensions/payment/processors/edupay.py

```diff
@@ -0,0 +1,384 @@
+        # Raise an exception for payments that were not accepted. Consuming code should be responsible for handling
+        # and logging the exception.
+        transaction_state = response['state_pol']
+
+        if transaction_state != self.TRANSACTION_ACCEPTED:
+            exception = {
+                self.TRANSACTION_DECLINED: TransactionDeclined,
+                self.TRANSACTION_ERROR: GatewayError,
+                self.TRANSACTION_PENDING: TransactionPending,
+            }.get(transaction_state, InvalidEdnxDecision)
+
+            raise exception
+
+        currency = response.get('currency')
+        total = Decimal(response.get('value'))
+        transaction_id = response.get('transaction_id')
+        card_number = response.get('cc_number', '')
+        card_type = response.get('lapPaymentMethod', '')
+
+        return HandledProcessorResponse(
+            transaction_id=transaction_id,
+            total=total,
+            currency=currency,
+            card_number=card_number,
+            card_type=card_type,
+        )
+
+    def _generate_signature(self, parameters, signature_type):
+        """
+        Sign the contents of the provided transaction parameters dictionary.
+
+        This allows PP to verify that the transaction parameters have not been tampered with
+        during transit.
+
+        We also use this signature to verify that the signature we get back from PP is valid for
+        the parameters that they are giving to us.
+
+        Arguments:
+            parameters (dict): A dictionary of transaction parameters.
+            signature_type: Digital signature created for each of the transactions
+
+        Returns:
+            unicode: the signature for the given parameters
+        """
+        signature_keys = []
+        extra_uncoded = ''
+        basic_uncoded = '{api_key}~{merchant_id}'.format(api_key=self.api_key, merchant_id=self.merchant_id)
+
+        if signature_type == self.PAYMENT_FORM_SIGNATURE:
+            signature_keys = self.ednx_configuration.get('SIGNATURE_PAYMENT_KEYS', [])
+        elif signature_type == self.CONFIRMATION_SIGNATURE:
+            # The logic to validate signatures on confirmation page:
+            # If the second decimal of the value parameter is zero, e.g. 150.00
+            # the parameter new_value to generate the signature should only have one decimal, as follows: 150.0
+            # If the second decimal of the value parameter is different from zero, e.g. 150.26
+            # the parameter new_value to generate the signature should have two decimals, as follows: 150.26
+            # taken from http://developers.payulatam.com/en/web_checkout/integration.html
+            # hence the parameter value is mandatory
+            value = parameters['value']
+            parameters['value'] = value[:-1] if value[-1] == '0' else value
+            signature_keys = self.ednx_configuration.get('SIGNATURE_CONFIRMATION_KEYS', [])
+
```

Comment:
what happens when is `else`?

Reply:
what do you mean exactly? when signature type isn't PAYMENT_FORM_SIGNATURE or CONFIRMATION_SIGNATURE?

Reply:
Yes, that is exactly the unresolved case here. What happens then?

Reply:
Talking to @andrey-canon is clear now. If the type is not accounted for here then in line 354 uncoded will be = to basic_uncoded which is desirable and normal in some configurations